module/VuFind/src/VuFind/ILS/Driver/KohaRest.php

```diff
@@ -1742,67 +1742,27 @@ protected function getItemStatusCodes($item)
-            foreach ($item['availability']['unavailabilities'] as $key => $reason) {
-                if (isset($this->itemStatusMappings[$key])) {
-                    $statuses[] = $this->itemStatusMappings[$key];
+            foreach ($item['availability']['unavailabilities'] as $code => $data) {
+                // If we have a direct mapping, use it:
+                if (isset($this->itemStatusMappings[$code])) {
+                    $statuses[] = $this->itemStatusMappings[$code];
                     continue;
                 }
-                $parts = explode('::', $key, 2);
-                $statusType = $parts[0];
-                $status = $parts[1] ?? '';
-
-                if ('Item' === $statusType || 'ItemType' === $statusType) {
-                    switch ($status) {
-                    case 'CheckedOut':
-                        $overdue = false;
-                        if (!empty($reason['due_date'])) {
-                            $duedate = $this->dateConverter->convert(
-                                'Y-m-d',
-                                'U',
-                                $reason['due_date']
-                            );
-                            $overdue = $duedate < time();
-                        }
-                        $statuses[] = $overdue ? 'Overdue' : 'Charged';
-                        break;
-                    case 'Lost':
-                        $statuses[] = 'Lost--Library Applied';
-                        break;
-                    case 'NotForLoan':
-                    case 'NotForLoanForcing':
-                        // NotForLoan is special: status has a library-specific
-                        // status number. Allow mapping of different status numbers
-                        // separately (e.g. Item::NotForLoan with status number 4
-                        // is mapped with key Item::NotForLoan4):
-                        $statusKey = $key . ($reason['status'] ?? '-');
-                        // Replace ':' in status key if used as status since ':' is
-                        // the namespace separator in translatable strings:
-                        $statuses[] = $this->itemStatusMappings[$statusKey]
-                            ?? $reason['code'] ?? str_replace(':', '_', $statusKey);
-                        break;
-                    case 'Transfer':
-                        $onHold = false;
-                        if (!empty($item['availability']['notes'])) {
-                            foreach (array_keys($item['availability']['notes'])
-                                as $noteKey
-                            ) {
-                                if ('Item::Held' === $noteKey) {
-                                    $onHold = true;
-                                    break;
-                                }
-                            }
+
+                // Check for a mapping method for the unavailability reason:
+                $methodName = 'GetStatusCode'
```

Comment:
Should this start with a lowercase 'g'? (I realize that method names are case-insensitive in PHP, but it just feels like it would be better to match them up).

Reply:
Yes, now fixed.